### ISSUE TITLE
Prerequisites for the Build and Deploy Your Own GitHub Copilot 2023 Universe Workshop 

### ISSUE THREAD

**Issue by camihmerhar** [githubuniverseworkshops/my-universe-copilot#1]
Hello Universe 2023 Workshopers 👋🏼!

We are so excited to see you all soon.

In preparation for our jam packed Build and Deploy Your Own GitHub Copilot workshop, we will need you to complete [the following prework, which can be found in our README](https://github.com/githubuniverseworkshops/my-universe-copilot#preparation)

Looking forward to seeing you at Universe!

@InfernapeXavier , @fabriguespe , @faucherc , @siddharta1337 , @acp9191 , @watsonjm , @darhill78 , @AndrewEverywhere , @teunis90 , @dkian214 , @brpowell , @jimseiwert , @duluca , @DidierRLopes , @jrthartwig , @LindaFay , @laurif , @lawanidriss , @anshitmishra , @christianotieno , @lautapercuspain , @cpcloud , @darshankhunt , @elviskahoro , @julioarruda , @imjwang , @Berryq460 , @voongc , @shaneakr , @Harla101 , @ahmad-luqman , @lexoskeleton , @twillmsbynder , @gerkovink , @RinatS , @samyukthagopalsamy , @gjdame , @tsavo-at-pieces , @mark-at-pieces , @ninjha01 , @jahAtFriends , @IgnacioPalma , @caithlinpeters , @MykalMachon , @mcdonaldts , @nfurnosanofi , @evebartl , @jfw225 , @dot-derek , @hepaestus

**Comment by camihmerhar** [comment]
Hello Universe 2023 Workshopers 👋🏼!

We are so excited to see you all soon.

In preparation for our jam packed Build and Deploy Your Own GitHub Copilot workshop, we will need you to complete [the following prework, which can be found in our README](https://github.com/githubuniverseworkshops/my-universe-copilot#preparation)

Looking forward to seeing you at Universe!

@dgajjarhzn , @kpc17 , @kjaymiller , @dzeyelid , @sutharzan-ch , @LuisDavidRodriguez , @saracope , @PatrickTulskie , @amaltson , @SterlingChin , @OzHache , @cogoken , @dariawhitecastle , @jhdub23 , @ansckasell , @tazarov , @djosephshz , @dsanchezcr , @angusjellis , @cvallejo , @unicornemma , @expo25 , @vshashkov , @mstokes , @SANKASU , @SP-DBAG , @DrOleson , @jayant-kumawat , @alexis-tadrowski , @celinaLind , @ooples , @ricrosales , @hideaki , @KEN10120 , @zj-landeed , @neilzhao-aws , @pedropaulovc , @neekipatel , @tjkiss , @aberfeldy , @mrhein , @sdbroscious , @udemonia , @hvvikram , @kerbybarrett , @bhavyageethika , @Jersonquinterolopez , @rongxin-liu , @EdAnuff , @chriswblake , @sogani-amit , @2good4hisowngood , @jayant-kumawat , @jpmcap , @samar1tan , @nettiquet , @neil-tan , @JesseAA , @celene-isip-sp , @githubuniverseworkshops/universe-workshop-build-and-deploy-your-own-copilot

**Comment by expo25** [comment]
@camihmerhar Hi! Are we required to complete ALL steps in the README, down to the `Clean-Up` section BEFORE coming to Universe??

**Comment by camihmerhar** [comment]
> the following prework, which can be found in our README

Howdy @expo25 👋🏼! Great question, and nope, just up to [Create OpenAI Key.](https://github.com/githubuniverseworkshops/my-universe-copilot#5-optional-create-openai-api-key) The rest we will cover during the Workshop at Universe

**Comment by dkian214** [comment]
Hi, I’m getting the following error message in my Codespace but Copilot for AI still seems to work. I joined the waitlist and still have the same error. Do I need to change anything?

@dkian214 ➜ /workspaces/Build-and-Deploy-Your-Own-GitHub-Copilot (main) $ eval "$(github-copilot-cli alias -- "$0")"
@dkian214 ➜ /workspaces/Build-and-Deploy-Your-Own-GitHub-Copilot (main) $ github-copilot-cli auth
Copy this code: 6ED4-BAEC

Then go to https://github.com/login/device, paste the code in and approve the access.
Failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.
@dkian214 ➜ /workspaces/Build-and-Deploy-Your-Own-GitHub-Copilot (main) $ which version of node is installed
/home/codespace/nvm/current/bin/node
@dkian214 ➜ /workspaces/Build-and-Deploy-Your-Own-GitHub-Copilot (main) $ ?? Which version of node is installed

──────────────────── Command ────────────────────

node --version

────────────────── Explanation ──────────────────

○ node is the Node.js runtime.
  ◆ --version specifies that we want to print the version of the runtime.

🕖  Hold on, executing commmand...
v20.8.1
@dkian214 ➜ /workspaces/Build-and-Deploy-Your-Own-GitHub-Copilot (main) $



Mahalo,


Ian Morada
Sr. Solutions Architect
DK Tech Solutions
808-937-0331
***@***.***

Use this Calendly link to put 30 min on my calendar.
https://calendly.com/ian-dktechsolutions/30min
[A blue and black logo  Description automatically generated with low confidence]

From: Cami Hernandez Merhar ***@***.***>
Date: Tuesday, October 31, 2023 at 8:52 AM
To: githubuniverseworkshops/my-universe-copilot ***@***.***>
Cc: Ian Morada ***@***.***>, Mention ***@***.***>
Subject: [githubuniverseworkshops/my-universe-copilot] Prerequisites for the Build and Deploy Your Own GitHub Copilot 2023 Universe Workshop (Issue #1)

Hello Universe 2023 Workshopers 👋🏼!

We are so excited to see you all soon.

In preparation for our jam packed Build and Deploy Your Own GitHub Copilot workshop, we will need you to complete the following prework, which can be found in our README<https://github.com/githubuniverseworkshops/my-universe-copilot#preparation>

Looking forward to seeing you at Universe!

@InfernapeXavier<https://github.com/InfernapeXavier> , @fabriguespe<https://github.com/fabriguespe> , @faucherc<https://github.com/faucherc> , @siddharta1337<https://github.com/siddharta1337> , @acp9191<https://github.com/acp9191> , @watsonjm<https://github.com/watsonjm> , @darhill78<https://github.com/darhill78> , @AndrewEverywhere<https://github.com/AndrewEverywhere> , @teunis90<https://github.com/teunis90> , @dkian214<https://github.com/dkian214> , @brpowell<https://github.com/brpowell> , @jimseiwert<https://github.com/jimseiwert> , @duluca<https://github.com/duluca> , @DidierRLopes<https://github.com/DidierRLopes> , @jrthartwig<https://github.com/jrthartwig> , @LindaFay<https://github.com/LindaFay> , @laurif<https://github.com/laurif> , @LawanIdriss<https://github.com/LawanIdriss> , @anshitmishra<https://github.com/anshitmishra> , @christianotieno<https://github.com/christianotieno> , @lautapercuspain<https://github.com/lautapercuspain> , @cpcloud<https://github.com/cpcloud> , @darshankhunt<https://github.com/darshankhunt> , @elviskahoro<https://github.com/elviskahoro> , @julioarruda<https://github.com/julioarruda> , @imjwang<https://github.com/imjwang> , @berryq460<https://github.com/berryq460> , @voongc<https://github.com/voongc> , @shaneakr<https://github.com/shaneakr> , @Harla101<https://github.com/Harla101> , @ahmad-luqman<https://github.com/ahmad-luqman> , @lexoskeleton<https://github.com/lexoskeleton> , @twillmsbynder<https://github.com/twillmsbynder> , @gerkovink<https://github.com/gerkovink> , @RinatS<https://github.com/RinatS> , @samyukthagopalsamy<https://github.com/samyukthagopalsamy> , @gjdame<https://github.com/gjdame> , @tsavo-at-pieces<https://github.com/tsavo-at-pieces> , @mark-at-pieces<https://github.com/mark-at-pieces> , @ninjha01<https://github.com/ninjha01> , @jahAtFriends<https://github.com/jahAtFriends> , @IgnacioPalma<https://github.com/IgnacioPalma> , @caithlinpeters<https://github.com/caithlinpeters> , @MykalMachon<https://github.com/MykalMachon> , @mcdonaldts<https://github.com/mcdonaldts> , @nfurnosanofi<https://github.com/nfurnosanofi> , @evebartl<https://github.com/evebartl> , @jfw225<https://github.com/jfw225> , @dot-derek<https://github.com/dot-derek> , @hepaestus<https://github.com/hepaestus> , @dgajjarhzn , @kpc17 , @kjaymiller , @dzeyelid , @sutharzan-ch , @LuisDavidRodriguez , @saracope , @PatrickTulskie , @amaltson , @SterlingChin , @OzHache , @cogoken , @dariawhitecastle , @jhdub23 , @ansckasell , @tazarov , @djosephshz , @dsanchezcr , @angusjellis , @cvallejo , @unicornemma , @expo25 , @vshashkov , @mstokes , @SANKASU , @SP-DBAG , @DrOleson , @jayant-kumawat , @alexis-tadrowski , @celinaLind , @ooples , @ricrosales , @hideaki , @KEN10120 , @zj-landeed , @neilzhao-aws , @pedropaulovc , @neekipatel , @tjkiss , @aberfeldy , @mrhein , @sdbroscious , @udemonia , @hvvikram , @kerbybarrett , @bhavyageethika , @Jersonquinterolopez , @rongxin-liu , @EdAnuff , @chriswblake , @sogani-amit , @2good4hisowngood , @jayant-kumawat , @jpmcap , @samar1tan , @nettiquet , @neil-tan , @JesseAA , @celene-isip-sp , @iSiddharth20 , @miquelduranfrigola , @ataylorme , @adarshadatta , @GoogML , @amogh-axp , @tyroneclay , @EdAnuff

**Comment by bhavyageethika** [comment]
Sorry but this is a large blob of text to process what's going on in this
thread. I will be unsubscribing. Thank you

**Comment by dkian214** [comment]
Not sure why it blobbed like that in David’s response. I sent a message that I’m having trouble with GitHub Copilot for CLI. I get an access error and need to join a waitlist but I am still able to execute a Copilot CLI test successfully. David responded with the same issue plus an issue with Azure auth. I will also say that I received an acceptance email to the waitlist but haven’t seen if anything has changed in my Codespace yet.

Mahalo,


Ian Morada
Sr. Solutions Architect
DK Tech Solutions
808-937-0331
***@***.***

Use this Calendly link to put 30 min on my calendar.
https://calendly.com/ian-dktechsolutions/30min
[A blue and black logo  Description automatically generated with low confidence]

From: bhavyageethika ***@***.***>
Date: Friday, November 3, 2023 at 5:17 AM
To: githubuniverseworkshops/my-universe-copilot ***@***.***>
Cc: Ian Morada ***@***.***>, Mention ***@***.***>
Subject: Re: [githubuniverseworkshops/my-universe-copilot] Prerequisites for the Build and Deploy Your Own GitHub Copilot 2023 Universe Workshop (Issue #1)
Sorry but this is a large blob of text to process what's going on in this
thread. I will be unsubscribing. Thank you

**Comment by camihmerhar** [comment]
Howdy @dkian214 and @sdbroscious 👋🏼!

So sorry for y'all's experience, let's see if we can get to the bottom of this. 

I'm tagging @stebje and @vfm2, your lovely Workshop Facilitators to see if we can address your issues.

**Comment by expo25** [comment]
I can confirm to have the same issue as the above two having no access to Copilot CLI or Azure. I also received emails confirming I had access.

**Comment by LindaFay** [comment]
Can we get clarification on the Copilot CLI? The main repo that is referenced in the pre-work (githubuniverseworkshops/workshop-build-deploy-your-own-copilot), which lists the pre-requisites says that Copilot CLI is a "nice to have". But this pre-work tells us to install it. (And I have same issue as others mentioned above). So is it required or just a nice to have?

**Comment by stebje** [comment]
Hi @LindaFay, I'm Steffen and will be joining you as one of the workshop facilitators next week, great to meet you! 👋 

### 💡 GitHub Copilot CLI questions/issues

Copilot CLI is listed as [`Nice-to-have` in the main workshop repo](https://github.com/githubuniverseworkshops/workshop-build-deploy-your-own-copilot?tab=readme-ov-file#nice-to-have) because it is not a _necessary_ pre-requisite for completing the workshop, and we're aware that some might not have access yet. It is however a plus to have access, since part of the workshop objective is to get familiarized with the GitHub Copilot CLI capabilities.

If you are able to run GitHub Copilot CLI commands and get the expected output then you should be good to go, I can't say for sure why you'd get a response saying that you don't have access. 

### 💡 Azure CLI auth questions/issues

For those who are having trouble authenticating with Azure, ensure that you already have an active Azure account as noted in [`Pre-requisites - Must-have`](https://github.com/githubuniverseworkshops/workshop-build-deploy-your-own-copilot?tab=readme-ov-file#mega-prerequisites). I cannot see any technical details/console output from folks trying to authenticate using `az login --use-device-code`, please provide that in this thread and we'll be better suited to support 🙏 

CC'ing others that have mentioned this topic above; @dkian214 @sdbroscious @expo25

**Comment by ooples** [comment]
I'm having the same issues

**Comment by watsonjm** [comment]
I am also having the same issue with no access to GitHub Copilot

**Comment by JesseAA** [comment]
:( i can't seem to get Azure to work, no matter what I do . I'm dropping out of workshop

**Comment by PatrickTulskie** [comment]
Same here. No access to the copilot CLI.

**Comment by stebje** [comment]
Hi @JesseAA can you post in this thread what you're seeing when trying to [authenticate with Azure](https://github.com/githubuniverseworkshops/my-universe-copilot?tab=readme-ov-file#4-authenticate-with-azure)?

Any details will help me provide support 🙏

**Comment by amogh-axp** [comment]
No access to GH copilot CLI. See error below
```
Then go to https://github.com/login/device, paste the code in and approve the access.
Failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.
```

**Comment by ataylorme** [comment]
Likewise no access to GitHub Copilot CLI

```
Failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.
```

A separate question: can we use the [Azure OpenAI API](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference)?

**Comment by ataylorme** [comment]
I created a free Azure account but am getting errors creating the deployment group

```
az deployment group create --resource-group "Universe2023CopilotRG" --template-file ./templates/template-appService.json --parameters ./templates/parameters-template-appService.json
{"code": "InvalidTemplateDeployment", "message": "The template deployment 'template-appService' is not valid according to the validation procedure. The tracking id is 'REDACTED'. See inner errors for details."}

Inner Errors: 
{"code": "ValidationForResourceFailed", "message": "Validation failed for a resource. Check 'Error.Details[0]' for more information."}

Inner Errors: 
{"code": "SubscriptionIsOverQuotaForSku", "message": "This region has quota of 0 instances for your subscription. Try selecting different region or SKU."}
```

**Comment by stebje** [comment]
👋 Hi @ataylorme, did you try and deploy the templates as-is, or with the changes needed as described in the "Important" section of the [instruction step](https://github.com/githubuniverseworkshops/my-universe-copilot?tab=readme-ov-file#4-create-an-azure-app-service-resource-in-the-resource-group-the-arm-template-is-located-in-templatestemplate-appservicejson-and-the-input-parameters-are-in-templatesparameters-template-appservicejson)? The template parameter file requires some changes before being able to successfully deploy it as described in that section.

**Comment by ataylorme** [comment]
@stebje I updated the fields as noted in the doc. If I am working too far ahead I can wait to troubleshoot this at the workshop

**Comment by stebje** [comment]
👋 Hi @ataylorme thank you for the feedback. Note that the deployment is **not** part of the preparation and we can address issues during the workshop as well.

**Comment by iSiddharth20** [comment]
I don't have access to Copilot CLI either and have not received any communication since I've joined the waitlist. Is there any temporary provision for Workshop participants to access Copilot CLI?

**Comment by jersonmorrow** [comment]
No access to Copilot CLI either! I joined the waitlist. Got authenticated with Azure with my account.

**Comment by iSiddharth20** [comment]
@githubuniverseworkshops/universe-workshop-build-and-deploy-your-own-copilot 

If you're facing the following error:

`failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.`


Here is the Solution : 

- Run these 2 Commands
`npm i -g @githubnext/github-copilot-cli`
`eval "$(github-copilot-cli alias -- "$0")"`

- DO NOT Run the following Command
`github-copilot-cli auth`

- Try running the following Command to Verify if everything works as desired
`?? Which version of node is installed`

Basically, DO NOT AUTHENTICATE Copilot-CLI as the Workshop Codespace already has Copilot-CLI ready for usage.

I hope this works for everyone. Thanks to [@nilzzzz19 ](https://github.com/nilzzzz19) for the pro-tip.

Another thing to note, when you open a new terminal in the Codespace or when you reconnect to Codespace, make sure to run the `eval "$(github-copilot-cli alias -- "$0")"` command, this will get the copilot-cli running again.

**Comment by ataylorme** [comment]
Confirming I had to upgrade my Azure account from free to pay as you go in order to do the `az deployment group create` command without getting the `{"code": "SubscriptionIsOverQuotaForSku", "message": "` error. It doesn't work on the Azure free tier

**Comment by expo25** [comment]
@ataylorme I can also confirm I did too. After it was all said & done I tried deleting both the azure resource group & account from my main azure directory in fear i would get charged. Neither the readme nor the workshop covered any of these issues with Azure authentication. It was a very poorly designed workshop to be completely transparent. I learned absolutely nothing I couldn't learn on my own and wanted a refund.

**Comment by amogh-axp** [comment]
Likewise about the error with azure saying unable to create resources in the specified region. Had to upgrade to the pay as you go tier to resolve that piece.
Let me know if anyone figured out a way to revert to the free tier as I couldn't find that option after hunting for a bit.
Thanks!
________________________________

**Comment by hepaestus** [comment]
I cannot get the `github-copilot-cli auth` command to finish successfully. I keep getting the error that I don't have access yet despite doing all the things.

```
@hepaestus ➜ /workspaces/github-my-universe-copilot (main) $ github-copilot-cli auth
Copy this code: 1234-5678

Then go to https://github.com/login/device, paste the code in and approve the access.
Failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.
```
I was added to the waitlist a while back as far as I know.

Thanks,
P.

**Comment by hepaestus** [comment]
> @githubuniverseworkshops/universe-workshop-build-and-deploy-your-own-copilot
> 
> If you're facing the following error:
> 
> `failed to authenticate: You do not have access to GitHub Copilot CLI. Please join the waitlist at https://githubnext.com/projects/copilot-cli/., Unable to verify waitlist access.`
> 
> Here is the Solution :
> 
> * Run these 2 Commands
>   `npm i -g @githubnext/github-copilot-cli`
>   `eval "$(github-copilot-cli alias -- "$0")"`
> * DO NOT Run the following Command
>   `github-copilot-cli auth`
> * Try running the following Command to Verify if everything works as desired
>   `?? Which version of node is installed`
> 
> Basically, DO NOT AUTHENTICATE Copilot-CLI as the Workshop Codespace already has Copilot-CLI ready for usage.
> 
> I hope this works for everyone. Thanks to [@nilzzzz19 ](https://github.com/nilzzzz19) for the pro-tip.
> 
> Another thing to note, when you open a new terminal in the Codespace or when you reconnect to Codespace, make sure to run the `eval "$(github-copilot-cli alias -- "$0")"` command, this will get the copilot-cli running again.

OK This has worked for me!

**Comment by stebje** [comment]
👋 Thank you to everyone who joined and contributed to the workshop ! 

Some asked for the slides so these have been added to the [workshop repo](https://github.com/githubuniverseworkshops/workshop-build-deploy-your-own-copilot?tab=readme-ov-file#books-resources) 👍
